src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -0,0 +1,457 @@
+                    {
+                        fields = new PemFields(labelRange, base64range, location, decodedBase64Size);
+                        return true;
+                    }
+                    break;
+                }
+                while (TryReadNextLine(pemData, ref postebLinePosition, out postebLineRange));
+            }
+
+            fields = default;
+            return false;
+        }
+
+        private static bool IsValidLabel(ReadOnlySpan<char> data)
+        {
+            static bool IsInRange(char c, char min) => (uint)(c - min) <= (uint)('\x7E' - min);
+
+            if (data.Length == 0)
+                return true;
+
+            // First character of label must be a labelchar, which is a character
+            // in 0x21..0x7e (both inclusive), except hyphens.
+            char firstChar = data[0];
+            if (!IsInRange(firstChar, '\x21') || firstChar == '-')
+                return false;
+
+            for (int index = 1; index < data.Length; index++)
+            {
+                // Characters after the first are permitted to be spaces and hyphens
+                if (!IsInRange(data[index], '\x20'))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        private static bool IsValidBase64(
+            ReadOnlySpan<char> data,
+            Range content,
+            out Range base64range,
+            out int decodedSize)
+        {
+            static bool IsBase64Char(char c) =>
+                (c >= '0' && c <= '9') ||
+                (c >= 'A' && c <= 'Z') ||
+                (c >= 'a' && c <= 'z') ||
+                c == '+' || c == '/' || c == '=';
+
+            int base64chars = 0;
+            int precedingWhiteSpace = 0;
+            int trailingWhiteSpace = 0;
+            (int offset, int length) = content.GetOffsetAndLength(data.Length);
+            for (int index = offset; index < offset + length; index++)
+            {
+                char c = data[index];
+
+                if (IsBase64Char(c))
+                {
+                    trailingWhiteSpace = 0;
+                    base64chars++;
+                }
+                else if (c == ' ' || c == '\n' || c == '\r' ||
+                         c == '\t' || c == '\v')
+                {
+                    if (base64chars == 0)
+                    {
+                        precedingWhiteSpace++;
+                    }
+                    else
+                    {
+                        trailingWhiteSpace++;
+                    }
+                }
+                else
+                {
+                    base64range = default;
+                    decodedSize = 0;
+                    return false;
+                }
+            }
+
+            base64range = (offset + precedingWhiteSpace)..(offset + (length - trailingWhiteSpace));
+            decodedSize = (base64chars * 3) / 4;
```

Comment:
I believe this give the incorrect results when the input span of char contains padding.
```C#
var r = new Random(42);
_bytes = new byte[1_000];
r.NextBytes(_bytes);
_base64String = Convert.ToBase64String(_bytes);
IsValidBase64(_base64String, out int decodedSize); // returns 1002, it should be 1000.
```

Reply:
Hm, thanks. I'll look at this, fix, and see overall what can be done about perf and correctness.